src/bootstrap/compile.rs

```diff
@@ -1051,9 +1051,17 @@ impl Step for Assemble {
     }
 
     fn make_run(run: RunConfig<'_>) {
-        run.builder.ensure(Assemble {
-            target_compiler: run.builder.compiler(run.builder.top_stage + 1, run.target),
-        });
+        // NOTE: `builder.compiler` has the side effect of running `Assemble`
```

Comment:
This comment seems a little confusing since there's no call to builder.compiler here (anymore), I think elaborating on it a little would be good.